src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,244 @@
+            throw new DataProcessingException("Cannot get all cars from db", e);
+        }
+
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET manufacturer_id = ? "
+                + "WHERE id = ? AND cars.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setLong(2, car.getId());
+            statement.executeUpdate();
+            deleteDrivers(car);
+            List<Driver> drivers = car.getDriverList();
+            if (drivers != null) {
+                for (Driver driver : drivers) {
+                    addDriverForCar(car, driver);
+                }
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Car cannot be update by id: " + car.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement
+                        = connection.prepareStatement(query)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Car cannot be deleted by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM cars_drivers cd"
+                + "JOIN cars c ON c.id = cd.car_id "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE driver_id = ? AND c.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement
+                         = connection.prepareStatement(query)) {
+            getAllStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all cars by drivers id: " + driverId, e);
+        }
+        addDriversForCars(cars);
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setManufacturer(getManufacturer(resultSet.getObject("manufacturer_id", Long.class)));
+        car.setId(resultSet.getObject("car_id", Long.class));
+        return car;
+    }
+
+    private Manufacturer getManufacturer(Long id) {
+        String query = "SELECT * FROM manufacturers WHERE id = "
```

Comment:
Because here I create a car object and I have no information about manufacturer in "cars" table and have only in "manufacturers" table. In get qurey for car I can get only id, which is used to find other info

Reply:
you're passing result set that contains all this information. what is `m.id`, `m.name` and `m.country` if not the info you seek. you can also add to this query m.is_deleted = FALSE
![image](https://user-images.githubusercontent.com/58442482/118609369-55262100-b7c3-11eb-9727-cc06b3dceb5b.png)
you can create new Manufacturer object in `private getCar()`, get all you need from result set and set the car with this object

Reply:
Ok, got it